src/middleware.ts

```diff
@@ -240,7 +240,8 @@ export const persist = <S extends State>(
 
     if (whitelist) {
       Object.keys(state).forEach((key) => {
-        !whitelist.includes(key) && delete state[key]
+        const keyTyped = key as keyof S
+        !whitelist.includes(keyTyped) && delete state[keyTyped]
```

Comment:
I would do this. Does it work?
```ts
      ;(Object.keys(state) as (keyof S)[]).forEach((key) => {
        !whitelist.includes(key) && delete state[key]
```

Reply:
Addressed in https://github.com/pmndrs/zustand/pull/376/commits/8473244cce1fa4a10d8a63cc977bae50508d0813